lib/spack/spack/cmd/env.py

```diff
@@ -86,6 +86,13 @@ def env_activate_setup_parser(subparser):
         const="bat",
         help="print bat commands to activate the environment",
     )
+    shells.add_argument(
+        "--pwsh",
+        action="store_const",
+        dest="shell",
+        const="pwsh",
+        help="print powershell commands to activate environment",
+    )
```

Comment:
You'll need to run `spack commands --update-completion` for unit tests to pass

Reply:
Addressed and pushed!